### PR TITLE
PynAdjust restructure

### PR DESCRIPTION
**Pull request by nicgowans**
### PynAdjust Restructure

Rationalisation and improvement of modules for consuming DynAdjust results

**Main changes:**

- 5 modules [adj.py, apu.py, xyz.py, pynadjust_classes.py, common_fn.py] condensed into results.py
- Structure streamlined, readability improved
- Most data structures now defined using class objects
- *.adj, *.apu, *.xyz files can now be consumed with a single call
- Alternatively, *.adj, *.apu, *.xyz files can be read independently with functions
- Previous functionality preserved

**New functionality:**

- Add variable type B uncertainties to stations (or subsets of stations)
- Recompute physical heights (and uncertainties) using an external ntv2 geoid model (e.g., for [Australian Vertical Working Surface heights](https://www.icsm.gov.au/australian-vertical-working-surface))
- Tally overall msr counts and msr-at-stns counts
- Coordinate comparisons
- Relative uncertainties can be computed by several strategies (many-many, nearest-neighbours, or specific pairs)

### CODE REVIEW

**Review by harry093 — APPROVED** [review]
Thanks for all the work on this, Nic, and thanks for the prompt to review, Josh. :-)

File: pynadjust/results.py [hunk 1]
```diff
@@ -0,0 +1,1989 @@
...
+                if m.msr_id in metadata:
+                    m.epoch = metadata[m.msr_id]['epoch']
+                    m.source = metadata[m.msr_id]['source']
+            else:
+                if m.msr_id:
+                    if m.msr_id[0] in metadata:
+                        m.epoch = metadata[m.msr_id[0]]['epoch']
+                        m.source = metadata[m.msr_id[0]]['source']
+
+    def write_shp(self, network_name):
```
Comment (BatchelorJ) [inline pynadjust/results.py]: Are you able to add an output directory argument to this method? When I tested it wrote the shapefiles directly into a folder inside the PynAdjust directory

**Review by BatchelorJ — CHANGES_REQUESTED** [review]
Hi @nicgowans , these changes all look super awesome to me! I've run some basic testing of the functionality but not gone too deep. The things I haven't looked into are the APU tools and the tools that require measurement IDs. Happy for this to go live and we can then get more familiar using it. The only change I'd suggest would be an argument in the `write_shp` method for pathing shapefiles into a specific directory. Cheers, Josh

File: pynadjust/results.py (hunk 1 above)
Comment (nicgowans) [reply pynadjust/results.py]: Thanks, @BatchelorJ. 

By default, the 'shp' directory is created within the current working directory, where ever that may be. 

As requested, I've added the option to specify where the shapefiles are written.

Enjoy!

**Review by BatchelorJ — APPROVED** [review]
Excellent stuff, thanks @nicgowans!

**Pull request merged**